### PR TITLE
Add `--max-results` option to mlflow experiments search

### PR DESCRIPTION
**Pull request by alkispoly-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/alkispoly-db/mlflow/pull/19359?quickstart=1)

#### Install mlflow from this PR

```
# mlflow
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/19359/merge
# mlflow-skinny
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/19359/merge#subdirectory=libs/skinny
```

For Databricks, use the following command:

```
%sh curl -LsSf https://raw.githubusercontent.com/mlflow/mlflow/HEAD/dev/install-skinny.sh | sh -s pull/19359/merge
```

</p>
</details>

### Related Issues/PRs

N/A

### What changes are proposed in this pull request?

This PR adds a `--max-results` option to the `mlflow experiments search` CLI command, allowing users to limit the number of experiments returned. The backend API already supports this parameter; this change exposes it through the CLI.

**Key features:**
- New `--max-results` CLI option with integer validation
- Validates non-negative values with clear error messages
- Default behavior unchanged (returns all experiments when not specified)
- Fully backward compatible

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [x] Manual tests

**New tests added:**
- Parametrized tests for default, explicit values, and combined flags
- Edge case test for zero value
- Validation test for negative values
- Parameter introspection test

**Manual testing:**
```bash
# Test with Databricks backend
export MLFLOW_TRACKING_URI="..."

# Returns exactly 5 experiments
mlflow experiments search --max-results 5

# Returns exactly 3 experiments  
mlflow experiments search --max-results 3

# Validates negative values
mlflow experiments search --max-results -5
# Error: Invalid value for '--max-results': max-results must be a non-negative integer
```

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

The CLI option is self-documenting through `--help` text.

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Added `--max-results` option to `mlflow experiments search` command to limit the number of experiments returned, improving performance and helping avoid rate limits when querying backends with many experiments.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows
- [ ] `area/gateway`: MLflow AI Gateway client APIs, server, and third-party integrations
- [ ] `area/prompts`: MLflow prompt engineering features, prompt templates, and prompt management
- [ ] `area/tracing`: MLflow Tracing features, tracing APIs, and LLM tracing functionality
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Review by harupy — APPROVED** [review]

File: tests/test_cli.py [hunk 1]
```diff
@@ -838,6 +838,53 @@ def test_env_file_loading_invalid_path() -> None:
...
+def test_experiments_search_max_results_negative():
+    result = CliRunner().invoke(
+        cli, ["experiments", "search", "--max-results", "-1"], catch_exceptions=False
+    )
+    assert result.exit_code != 0
+    assert "max-results must be a non-negative integer" in result.output
+
+
+def test_experiments_search_command_params():
+    from mlflow.experiments import commands
```
Comment (harupy) [inline tests/test_cli.py]: do we need to import this here?

Comment (harupy) [reply tests/test_cli.py]: btw do we need this test? Seems useless

File: mlflow/experiments.py [hunk 2]
```diff
@@ -55,12 +62,19 @@ def create(experiment_name, artifact_location):
     help="Select view type for experiments. Valid view types are "
     "'active_only' (default), 'deleted_only', and 'all'.",
 )
-def search_experiments(view):
+@click.option(
+    "--max-results",
```
Comment (harupy) [inline mlflow/experiments.py:66]: does the output have a token to fetch the next page?

Comment (alkispoly-db) [reply mlflow/experiments.py:66]: Not at this point -- let's introduce this later as needed.

File: tests/test_cli.py (hunk 1 above)
Comment (alkispoly-db) [reply tests/test_cli.py]: Agreed that it's not high value. I will remove.

File: mlflow/experiments.py (hunk 2 above)
Comment (harupy) [reply mlflow/experiments.py:66]: sounds good

**Review by harupy — APPROVED** [review]
